### ISSUE TITLE
Add support for http+unix:// to the on-board http client

### ISSUE THREAD

**Issue by racktear** [tarantool/tarantool#3040]
The http+unix:// is http protocol over unix domain sockets. It doesn't have an established URL scheme, but implementations I've come across, do it by url-encoding the path to unix socket and putting it instead of the host name.

Example:

```
http+unix://%2Fvar%2Frun%2Fdocker.sock/v1.33/containers/json?limit=-1&size=0&trunc_cmd=1&all=1
```

The request should go to `/var/run/docker.sock`, with URL parameter in the http request set to `http://localhost/v1.33/containers/json?limit=-1&size=0&trunc_cmd=1&all=1`.

It is important, that "http+unix://" scheme is only for our http client to distinguish the intention to use unix sockets, and shouldn't creep into the actual http request, as the server likely doesn't understand it.

To test the implementation, you may issue a GET request to the URL above (provided you have Docker on your machine)

**Comment by racktear** [comment]
Just in case, here's the PoC version:

https://gist.github.com/racktear/fbf1c7b1ff2b6fc4acc171e11ea8e2bc

**Comment by rtsisyk** [comment]
Please support both "http+unix://path-to-file" and "http://unix/:/path/to/file"

**Comment by racktear** [comment]
@rtsisyk how do you differentiate the second variant from a valid URL?

**Comment by kbelyavs** [comment]
We can have an array of specific prefixes:
const char *const prefixes[] = {
		"http+unix://",
		"http://unix/:/",
	};
and just check in loop:
if (strncmp(url, prefixes[i], strlen(prefixes[i])) == 0) { ... }

**Comment by racktear** [comment]
@kbelyavs try parsing the url "http://unix/:/path/to/file" by our URL parser and see how it will parse the components. It is not even a valid URL.

**Comment by kbelyavs** [comment]
@racktear Yes, I see. It's decoded completely different way, but with code, based on your initial proposal it works fine.
@rtsisyk Could you provide complete schema for http://unix/:/path/to/file or give us some examples?

**Comment by kbelyavs** [comment]
Second type query is something like this: http://unix:/var/run/docker.sock:/v1.24/images/json in this schema socket isn't encoded and our library can't distinguish socket and path in this case. So each specific case should be solved its own way.

**Comment by racktear** [comment]
@kbelyavs I don't think we should support the case you mention. It's not a valid URI. @rtsisyk please consider it seriously.

**Comment by kbelyavs** [comment]
Agreed to use Yaroslav's version with compile time warning if UNIX_SOCKETS are not supported in installed curl version and also a runtime check.

**Comment by locker** [comment]
Fixed by 04e75f2c99d7b28299463830c8131391e7558586
